### PR TITLE
Main program logic

### LINKED ISSUE #29: Code out run method for proper functioning of program
Call the necessary methods from other classes and initialise classes so that program can start and run smoothly.

### PR DESCRIPTION
**Pull request by KVignesh122**
Fix #29

### CODE REVIEW

File: src/main/java/seedu/mtracker/MTracker.java [hunk 1]
```diff
@@ -14,24 +16,33 @@ public MTracker() {
...
-        Command command;
-        TextUi.greetAtStartUp();
-        String userInput = InputParser.getUserInput();
-        String[] commandComponents = InputParser.getCommandComponents(userInput);
-        try {
-            command = InputParser.filterByCommandType(commandComponents);
-            command.execute();
-        } catch (Exception e) {
-            TextUi.showErrorMessage(e);
+        Command command = new InvalidCommand();
```
Comment (theodorekwok) [inline src/main/java/seedu/mtracker/MTracker.java:19]: Try not to initialise command to invalid at the start, it works in the program wise but its confusing to the reader
https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#don-t-recycle-variables-or-parameters

Rather initialise command in the try catch block. I know you have a while loop so use a do while loop instead and in the catch block initialise the command as invalidCommand

File: src/main/java/seedu/mtracker/MTracker.java [hunk 2]
```diff
@@ -14,24 +16,33 @@ public MTracker() {
...
+        String userInput;
+        String[] commandComponents;
+        // Quit program after ExitCommand executed.
```
Comment (theodorekwok) [inline src/main/java/seedu/mtracker/MTracker.java:22]: Remove the comment

File: src/main/java/seedu/mtracker/console/InputParser.java [hunk 3]
```diff
@@ -24,6 +24,7 @@
     protected static Scanner inputScanner = new Scanner(System.in);
 
     public static String getUserInput() {
+        System.out.print("mTracker$>");
```
Comment (theodorekwok) [inline src/main/java/seedu/mtracker/console/InputParser.java:27]: Abstract this into TextUi class

File: src/main/java/seedu/mtracker/instrument/InstrumentManager.java [hunk 4]
```diff
@@ -5,7 +5,7 @@
 public class InstrumentManager {
 
     private static InstrumentManager instrumentManager;
-    private final ArrayList<Instrument> instruments;
+    private static ArrayList<Instrument> instruments;
```
Comment (theodorekwok) [inline src/main/java/seedu/mtracker/instrument/InstrumentManager.java:8]: Is there a reason why this was changed to static? It should an instance member and not a class member

File: src/main/java/seedu/mtracker/MTracker.java (hunk 1 above)
Comment (KVignesh122) [reply src/main/java/seedu/mtracker/MTracker.java:19]: OK that is smart.

File: src/main/java/seedu/mtracker/MTracker.java (hunk 2 above)
Comment (KVignesh122) [reply src/main/java/seedu/mtracker/MTracker.java:22]: Ok removed.

File: src/main/java/seedu/mtracker/console/InputParser.java (hunk 3 above)
Comment (KVignesh122) [reply src/main/java/seedu/mtracker/console/InputParser.java:27]: Done

File: src/main/java/seedu/mtracker/instrument/InstrumentManager.java (hunk 4 above)
Comment (theodorekwok) [reply src/main/java/seedu/mtracker/instrument/InstrumentManager.java]: Final are for variables that are assigned only once, not necessarily for constants only, so the naming still stays

**Pull request merged**